test/common/router/config_impl_speed_test.cc

```diff
@@ -132,13 +142,260 @@ static void bmRouteTableSizeWithExactPathMatch(benchmark::State& state) {
  *
  * This represents common OpenAPI path templating.
  */
-static void bmRouteTableSizeWithRegexMatch(benchmark::State& state) {
-  bmRouteTableSize(state, RouteMatch::PathSpecifierCase::kSafeRegex);
+static void bmRouteTableSizeWithRegexMatchFixPathLen(benchmark::State& state) {
+  bmRouteTableSize(state, RouteMatch::PathSpecifierCase::kSafeRegex,
+                  "^/shelves/[^\\\\/]+/route_", "/shelves/shelf_");
+}
+
+BENCHMARK(bmRouteTableSizeWithPathPrefixMatchFixPathLen)->RangeMultiplier(2)->Ranges({{1, 2 << 13}});
+BENCHMARK(bmRouteTableSizeWithExactPathMatchFixPathLen)->RangeMultiplier(2)->Ranges({{1, 2 << 13}});
+BENCHMARK(bmRouteTableSizeWithRegexMatchFixPathLen)->RangeMultiplier(2)->Ranges({{1, 2 << 13}});
+
+// Generate path rule prefix based on the pattern 'abcdefghij', and path_len
+static std::string GetPrefixPathRuleFromBase(int path_len) {
+  std::string path_rule_base = "abcdefghij"; // pattern
+  std::string path_rule = "";
+
+  // repeating the pattern path_len/10 times
+  for (int repeat = 0; repeat < path_len / 10;  ++repeat) {
+    absl::StrAppend(&path_rule, path_rule_base);
+  }
+  return path_rule;
+}
+
+/**
+ * Benchmark a route table with prefix path matchers with different path rule length:
+ * - /abcdefghij...<repeat n times>_n
+ * - etc.
+ */
+static void bmRouteTableSizeWithPathPrefixMatchVaryPathLen(benchmark::State& state) {
+  std::string path_rule = GetPrefixPathRuleFromBase(state.range(1));
+  bmRouteTableSize(state, RouteMatch::PathSpecifierCase::kPrefix,
+                  path_rule, path_rule);
+}
+
+// Arg1 is the path rule table size. Arg2 is the path rule length.
+BENCHMARK(bmRouteTableSizeWithPathPrefixMatchVaryPathLen)
+    ->ArgPair(1, 10)
```

Comment:
There must be a easier way to specify these args without listing them all out?

Reply:
Yeah, this is cross product.

Reply:
There is a RangePair() libraray as below can do this. However, the code will randomly select the numbers from the low/high range, and there is less control.  In my case, I want specifically benchmarking certain number of rules, like 50, 500, 1000, 10000, also certain rule length, like 100, 500, 1000. Benchmarking these numbers are specially required by design. And ArgPair() provides the control for us thus choosing it.

 // Pick a set of values A from the range [lo1..hi1] and a set
  // of values B from the range [lo2..hi2].  Run the benchmark for
  // every pair of values in the cartesian product of A and B
  // (i.e., for all combinations of the values in A and B).
  // REQUIRES: The function passed to the constructor must accept arg1,arg2.
  Benchmark* RangePair(int lo1, int hi1, int lo2, int hi2);

Reply:
I think `ArgsProduct` might do what you want here.